#! /bin/sh -e
## 90_armeabi.dpatch.dpatch by Lennert Buytenhek <buytenh@wantstofly.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Add support for EABI ARM platforms to Qt.

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@
--- qt4-x11-4.2.1/src/corelib/global/qglobal.h.orig	2007-01-04 23:57:34.000000000 +0000
+++ qt4-x11-4.2.1/src/corelib/global/qglobal.h	2007-01-04 23:58:06.000000000 +0000
@@ -290,7 +290,7 @@
 #  if defined(Q_OS_DARWIN) && __GNUC__ == 3 && (__GNUC_MINOR__ >= 1 && __GNUC_MINOR__ < 3)
 #    define Q_BROKEN_DEBUG_STREAM
 #  endif
-#  if (defined(__arm__) || defined(__ARMEL__)) && !defined(QT_MOC_CPP)
+#  if (defined(__arm__) || defined(__ARMEL__)) && !defined(__ARM_EABI__) && !defined(QT_MOC_CPP)
 #    define Q_PACKED __attribute__ ((__packed__))
 #    define Q_NO_PACKED_REFERENCE
 #  endif
